parser_test.go

```diff
@@ -2040,3 +2060,32 @@ func TestContainsUnescaped(t *testing.T) {
 		}
 	}
 }
+
+func TestValidNetworks(t *testing.T) {
+	for _, tt := range []struct {
+		strs []string
+		want bool
+	}{
+		{
```

Comment:
Please add a descriptive name to each test and include in the failure message, so that if the test fails it is clear exactly which piece is broken.

Reply:
"all valid" "double negation" "invalid variable" "invalid CIDR" etc
Please also add some IPv6 test cases.

Reply:
done